File: starter_kits/C#/Core/Dropoff.cs
namespace Halite3.Core
{
    public class Dropoff : Entity
    {
        public Dropoff(PlayerId owner, EntityId id, Position position) :
            base(owner, id, position)
        {
        }

        internal static Dropoff _generate(PlayerId playerId)
        {
            Input input = Input.readInput();

            EntityId dropoffId = new EntityId(input.getInt());
            int x = input.getInt();
            int y = input.getInt();

            return new Dropoff(playerId, dropoffId, new Position(x, y));
        }
    }
}

File: starter_kits/TypeScript/hlt/Dropoff.ts
import { Position } from "./Position";
import { Entity } from './entity';

/** Represents a dropoff. */
export class Dropoff extends Entity {
    /**
     * Create a Dropoff for a specific player from the engine input.
     */
    static async _generate(playerId: number, getLine: () => Promise<string>) {
        const [id, xPos, yPos] = (await getLine())
            .split(/\s+/)
            .map(x => parseInt(x, 10));
        return { id, dropoff: new Dropoff(playerId, id, new Position(xPos, yPos)) };
    }
}